### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by hufftheweevil** [caseyjhol/node-red-contrib-tplink#31]
There is STILL a memory leak somehow. When the flows are redeployed or restarted, old versions of the client connection remain in tact and polling continues for them as well. I'm not sure why yet, since this bit of code
```
    node.on('close', function () {
      // NOTE: We reference node.client.devices here, just in case
      // there were devices initialized on this client that did not
      // make it into the node.devices map. This ensures no memory
      // leaking in case of bugs/errors.
      node.client.devices.forEach(device => {
        device.stopPolling()
        device.closeConnection()
      })
      node.client.stopDiscovery()
    })
```
was supposed to prevent that from happening. I'm working on a fix right now.

**Comment by hufftheweevil** [comment]
It appears the close event is not getting triggered. I've got a console log inside the function and it never logs. I've tried all variations: `this.on('close', ... )`, `node.on('close', ... )`, `function()`, `function(done)`, `function(removed, done)`. Just to prove I'm not going crazy, I even tried using the `done` callback and delaying it with `setTimeout(done, 3000)` and there is no delay in redeployment.

Has anyone else experienced this? Any ideas?
